vllm/entrypoints/openai/serving_engine.py

```diff
@@ -793,23 +819,12 @@ async def _preprocess_chat(
         )
         _chat_template_kwargs.update(chat_template_kwargs or {})
 
-        request_prompt: Union[str, list[int]]
-
-        if tokenizer is None:
-            request_prompt = "placeholder"
```

Comment:
we could still do the non-tokenizer case inline to avoid the await barrier in this case...